vignettes/scoring-rules.Rmd

```diff
@@ -0,0 +1,470 @@
+
+$$\text{ape} = \frac{|y - \hat{y}|}{|y|}$$
+
+The absolute percentage error is only an appropriate rule if $\hat{y}$ corresponds to the $\beta$-median of the forecaster's predictive distribution with $\beta = -1$. The $\beta$-median, $\text{med}^{(\beta)}(F)$, is the median of a random variable whose density is proportional to $y^\beta f(y)$. The specific $\beta$-median that corresponds to the absolute percentage error is $\text{med}^{(-1)}(F)$. Otherwise, results will be misleading (see @gneitingMakingEvaluatingPoint2011). 
+
+---
+
+# Binary forecasts
+
+See a list of the default metrics for point forecasts by calling `?metrics_binary()`. 
+
+This is an overview of the input and output formats for point forecasts: 
+
+```{r, echo=FALSE, out.width="100%", fig.cap="Input and output formats: metrics for binary forecasts."}
+knitr::include_graphics(file.path("scoring-rules", "input-binary.png"))
+```
+
+
+## Brier score
+
+**Observation**: $y$, either 0 or 1
+
+**Forecast**: $p$, a probability that the observed outcome will be 1. 
+
+The Brier score is a strictly proper scoring rule. It is computed as the mean squared error between the probabilistic prediction and the observed outcome.
+
+\begin{equation}
+    \text{BS}(p, y) = (p - y)^2 = 
+    \begin{cases}
+        p^2,       & \text{if } y = 1\\
+        (1 - p)^2,   & \text{if } y = 0
+    \end{cases}
+\end{equation}
+
+The Brier score and the logarithmic score (see below) differ in how they penalise over- and underconfidence (see @macheteContrastingProbabilisticScoring2012). The Brier score penalises overconfidence and underconfidence in probability space the same. Consider the following example: 
+
+```{r}
+n <- 1e6
+p_true <- 0.7
+observed <- factor(rbinom(n = n, size = 1, prob = p_true), levels = c(0, 1))
+
+p_over <- p_true + 0.15
+p_under <- p_true - 0.15
+
+abs(mean(brier_score(observed, p_true)) - mean(brier_score(observed, p_over)))
+abs(mean(brier_score(observed, p_true)) - mean(brier_score(observed, p_under)))
+```
+
+See `?brier_score()` for more information.
+
+
+## Logarithmic score
+
+**Observation**: $y$, either 0 or 1
+
+**Forecast**: $p$, a probability that the observed outcome will be 1. 
+
+The logarithmic score (or log score) is a strictly proper scoring rule. It is computed as the negative logarithm of the probability assigned to the observed outcome. 
+
+\begin{equation}
+    \text{Log score}(p, y) = - \log(1 - |y - p|) = 
+    \begin{cases}
+        -\log (p),       & \text{if } y = 1\\
+        -\log (1 - p),   & \text{if } y = 0
+    \end{cases}
+\end{equation}
+
+The log score penalises overconfidence more strongly than underconfidence (in probability space). Consider the following example: 
+
+```{r}
+abs(mean(logs_binary(observed, p_true)) - mean(logs_binary(observed, p_over)))
+abs(mean(logs_binary(observed, p_true)) - mean(logs_binary(observed, p_under)))
+```
+
+See `?logs_binary()` for more information.
+
+
+---
+
+# Sample-based forecasts
```

Comment:
Need to talk about different kinds of probabilistic forecasts in the intro as currently this is not explained.